python/src/worker/tasks/differential_expression.py

```diff
@@ -112,12 +113,19 @@ def compute(self):
         if len(second_cell_set) == 0:
             raise Exception("No cells id fullfills the 2nd cell set.")
 
-        # create request from the corresponding marked cells
         request = {
             "baseCells": [int(x) for x in first_cell_set],
             "backgroundCells": [int(x) for x in second_cell_set],
+            "pagination": self.pagination,
         }
 
+        if "filters" in self.pagination:
+            gene_filter = self.pagination["filters"][0]["expression"]
+            regex_chars = "{}|()?¿*+|/.<>"
```

Comment:
I think we should probably handle all of the filtering logic in one single part, either the R worker or the python worker. I am not sure about this, but do you think it might be a good option to handle all of the filtering in python?

Alternatively, we could handle all of the filtering logic in R, I would send the whole thing (the whole "filters" object) to R and handle all of it there to avoid splitting the management of the filtering between python and R.

Reply:
Completely agree with Martin here, otherwise we risk duplicating efforts in both python and R and ending up with an unmaintainable code